src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.PooledConnection.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Execution;
+using Microsoft.CodeAnalysis.Remote;
+
+namespace Microsoft.VisualStudio.LanguageServices.Remote
+{
+    internal sealed partial class ServiceHubRemoteHostClient : RemoteHostClient
+    {
+        private partial class ConnectionManager
+        {
+            private class PooledConnection : Connection
+            {
+                private readonly ConnectionManager _pools;
+                private readonly string _serviceName;
+                private readonly JsonRpcConnection _connection;
+
+                public PooledConnection(ConnectionManager pools, string serviceName, JsonRpcConnection connection)
+                {
+                    _pools = pools;
+                    _serviceName = serviceName;
+                    _connection = connection;
+                }
+
+                public override Task SetConnectionStateAsync(PinnedRemotableDataScope scope) =>
+                    _connection.SetConnectionStateAsync(scope);
+
+                public override Task InvokeAsync(string targetName, IReadOnlyList<object> arguments, CancellationToken cancellationToken) =>
+                    _connection.InvokeAsync(targetName, arguments, cancellationToken);
+
+                public override Task<T> InvokeAsync<T>(string targetName, IReadOnlyList<object> arguments, CancellationToken cancellationToken) =>
+                    _connection.InvokeAsync<T>(targetName, arguments, cancellationToken);
+
+                public override Task InvokeAsync(
+                    string targetName, IReadOnlyList<object> arguments,
+                    Func<Stream, CancellationToken, Task> funcWithDirectStreamAsync, CancellationToken cancellationToken) =>
+                    _connection.InvokeAsync(targetName, arguments, funcWithDirectStreamAsync, cancellationToken);
+
+                public override Task<T> InvokeAsync<T>(
+                    string targetName, IReadOnlyList<object> arguments,
+                    Func<Stream, CancellationToken, Task<T>> funcWithDirectStreamAsync, CancellationToken cancellationToken) =>
+                    _connection.InvokeAsync<T>(targetName, arguments, funcWithDirectStreamAsync, cancellationToken);
+
+                protected override void OnDisposed() =>
```

Comment:
💡 File a follow-up issue to convert this to a proper dispose pattern

Reply:
sure 
https://github.com/dotnet/roslyn/issues/24837

Reply:
Why not just fix it now?

Reply:
Also, why is Connection an abstract class vs. just an interface? It seems like this disposal pattern is only making things weird here.